package net.petemc.daycount.event;

import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.client.event.*;
import net.minecraftforge.eventbus.api.listener.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.petemc.daycount.*;
import net.petemc.daycount.client.*;
import net.petemc.daycount.util.*;


public class ClientEvents {

    @Mod.EventBusSubscriber(modid = DayCount.MOD_ID, value = Dist.CLIENT)
    public static class ClientForgeEvents {
        @SubscribeEvent
        public static void onKeyInput(InputEvent.Key event) {
            if (KeyBinding.DAYCOUNT_KEY.consumeClick()) {
                DayCount.dayCountEnabled = !DayCount.dayCountEnabled;
            }
        }

        @SubscribeEvent
        public static void registerGuiOverlays(CustomizeGuiOverlayEvent.Chat event) {

            DayCountHud.DAY_COUNT_HUD_INSTANCE.render(event.getGuiGraphics(), null);
        }

        @SubscribeEvent
        public static void onKeyRegister(RegisterKeyMappingsEvent event) {
            event.register(KeyBinding.DAYCOUNT_KEY);
        }
    }
}
